### PR TITLE
Materialize train test

### LINKED ISSUE #470: dataset materialize for train and test data separately
The current example in the doc materialize the train and test before doing split. Could this lead to potential data leakage, as the col_stats is calculated on all data, but not training data. And for production test, should we have a separate function to materialise the data from raw data frame? 

from torch_frame.datasets import Yandex
from torch_frame.data import DataLoader

dataset = Yandex(root='/tmp/adult', name='adult')
dataset.materialize()
dataset.shuffle()
train_dataset, test_dataset = dataset[:0.8], dataset[0.80:]
train_loader = DataLoader(train_dataset.tensor_frame, batch_size=128,
                        shuffle=True)
test_loader = DataLoader(test_dataset.tensor_frame, batch_size=128)

### PR DESCRIPTION
**Pull request by HoustonJ2013**
Closes #470.

Add the col_stats in the dataset materialize function, allowing the column stats calculated from train dataset to be used in the test data materialization.

```python
train_df = ..
test_df = ..
train_dataset = Dataset(train_df, 
                 col_to_stype=col_to_stype, 
                 target_col="target")
train_dataset.materialize()
test_dataset = Dataset(test_df, 
                 col_to_stype=col_to_stype, 
                 target_col="target")
test_dataset.materialize(col_stats=train_dataset.col_stats)
```

### CODE REVIEW

File: torch_frame/data/dataset.py [hunk 1]
```diff
@@ -589,23 +594,27 @@ def materialize(
...
+                # such that we do not accidentally swap labels in binary
+                # classification tasks.
+                if col == self.target_col and stype == torch_frame.categorical:
+                    index, value = self._col_stats[col][StatType.COUNT]
+                    if len(index) == 2:
+                        ser = pd.Series(index=index, data=value).sort_index()
+                        index, value = ser.index.tolist(), ser.values.tolist()
+                        self._col_stats[col][StatType.COUNT] = (index, value)
+        else:
+            self._col_stats = col_stats
```
Comment (weihua916) [inline torch_frame/data/dataset.py:617]: Can we add basic validation here?

Comment (HoustonJ2013) [reply torch_frame/data/dataset.py:627]: Add as suggested, please check

Comment (weihua916) [reply torch_frame/data/dataset.py:627]: Thanks!

**Review by weihua916 — APPROVED** [review]
